webview/src/experiments/components/table/body/Row.tsx

```diff
@@ -68,31 +67,29 @@ export const RowContent: React.FC<
     }
   }, [subRows, selectedRows])
 
-  const [menuActive, setMenuActive] = useState<boolean>(false)
-
   const running = isRunning(status)
   const queued = isQueued(status)
   const unselected = selected === false
   const isOdd = index % 2 !== 0 && !isRowSelected
 
+  const [hideOnClick, setHideOnClick] = useState<(() => void) | undefined>(
```

Comment:
I would use a boolean `shouldHide`, pass `setShouldHide` to `RowContextMenu` and pass `shouldHide` to the `ContextMenu`.  `ContextMenu` should handle its behaviour. (See further comments as to why this would be better).